src/dd/dd.rs

```diff
@@ -0,0 +1,745 @@
+#![crate_name = "uu_dd"]
+
+#[macro_use]
+extern crate chan;
+extern crate chan_signal;
+
+#[macro_use]
+extern crate uucore;
+
+use std::cmp;
+use std::env;
+use std::fmt;
+use std::fs::{self, File, OpenOptions};
+use std::io::{self, Read, Seek, Write};
+use std::process::exit;
+use std::result;
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+use chan_signal::Signal;
+
+const BUF_SIZE: usize = 4096;
+
+pub struct Conf {
+    inp: Option<InpFileWrap>,
+    out: Option<OutFileWrap>,
+    inp_name: String,
+    out_name: String,
+    is_stdin: bool,
+    is_stdout: bool,
+
+    inp_block_size: usize,
+    out_block_size: usize,
+    cnv_block_size: usize,
+
+    skip: usize,
+    seek: usize,
+
+    count: usize,
+
+    inp_part: usize,
+    inp_full: usize,
+    out_part: usize,
+    out_full: usize,
+    trunc: usize,
+
+    flags: u32,
+
+    short_2_short: bool,
```

Comment:
I think I would prefer if `2` was `to`, but I'll leave it up to you.

Reply:
Did.